admin-ui/src/Meals/service.ts

```diff
@@ -0,0 +1,77 @@
+
+import { ApolloClient, gql } from '@apollo/client';
+
+const searchStringQuery = gql`
+  query SearchStrings($searchString: String!) {
+    query {
+      meals(
+        filter: {
+          or: [
+            { code: { includes: $searchString } }
+            { nameEn: { includes: $searchString } }
+            { nameFr: { includes: $searchString } }
+            { descriptionEn: { includes: $searchString } }
+            { descriptionFr: { includes: $searchString } }
+            { tags: { contains: [$searchString] } }
+          ]
+        }
+      ) {
+        edges {
+          node {
+            rowId
+          }
+        }
+      }
+    }
+  }
+`;
+
+const searchStringCategoryQuery = gql`
```

Comment:
category search need not be done seperately. I researched and found that 'overlaps:' works with enum.
I have put in an example query here for your easy understanding.
```
query searchMeal {
  meals (filter:{
    or:[
      {nameEn:{includesInsensitive:"coriander"}},
      {categories:{overlaps:DINNER}}
    ]
  }) {
    edges {
      node {
        nameEn
        categories
        rowId
      }
    }
  }
}

Reply:
That is because you are trying to pass the same string type. ENUM has to be of the type CategoryT. So I tried it this way and it works

```
query searchMeal($searchString: String, $category: [CategoryT]) {
  meals (filter:{
    or:[
      {nameEn:{includesInsensitive: $searchString}},
      {categories:{overlaps:$category}}
    ]
  }) {
    edges {
      node {
        nameEn
        categories
        rowId
      }
    }
  }
}
```
Under Query Variables
```
{
  "category": ["DINNER"],
  "searchString": "coriander"
}
```

Reply:
Finished implementing the one single query to fetch the database. Thank you again Shanthi.